File: api/src/main/java/com/api/service/EmployeeService.java
package com.api.service;

import com.api.dto.EmployeeDTO;
import com.api.model.Employee;

import java.util.List;

public interface EmployeeService {

    /**
     * Στο Employee θα αφήσω και το create/update εκτός από το save, για να υπάρχει κάπου η υλοποίηση */

    Employee dtoToEntity(EmployeeDTO employeeDTO);

    List<EmployeeDTO> findAll();

    Employee get(Long id);

    Employee create(EmployeeDTO employeeDTO);

    Employee update(EmployeeDTO employeeDTO);

    void delete(Long id);

    Employee save(EmployeeDTO employeeDTO);
}

File: front-end/src/app/em/employee/employee.service.js
(function() {

    angular.module('app.em.employee').service('employeeService', employeeService);

    function employeeService(employeeFactory) {
        return {
            getEmployees : getEmployees,
            getEmployee : getEmployee,
            saveEmployee : saveEmployee,
            deleteEmployee : deleteEmployee,
        }
        
        function getEmployees() {
            return employeeFactory.list({});
        }

        function getEmployee(id) {
            return employeeFactory.get({id},null);
        }

        function saveEmployee(Employee) {
            return employeeFactory.save({}, Employee);
        }

        function deleteEmployee(id) {
            return employeeFactory.delete({id},null);
        }
    }

})();